planner/core/rule_column_pruning.go

```diff
@@ -121,11 +121,6 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 				selfUsedCols = append(selfUsedCols, cols...)
 			}
 		}
-		// If all the group by items are pruned, we should add a constant 1 to keep the correctness.
```

Comment:
So `select count(*) from t` is equivalent to `select count(*) from t group by 1`?

Reply:
`select count(*) from t group by 1` is indeed different from `select count(*) from t`. But it can't be the reason to keep **THIS** `1` here. The former `1` represents the first column which is handling well somewhere, while the latter represents a constant could be eliminated.